### ISSUE TITLE
Odoo and PostgreSQL-ha: User bn_odoo already exists

### ISSUE THREAD

**Issue by joergfelder** [bitnami/charts#6328]
On my Kubernetes cluster I create a new namespace for PostgreSQL-ha and Odoo with 

```
helm upgrade --install postgresql-ha bitnami/postgresql-ha \
--values ~/odoo/values-postgres-ha.yaml \
--version 6.6.2 \
--namespace odoo \
--create-namespace
```

and 

```
helm upgrade --install odoo bitnami/odoo \
  --values ~/odoo/values-odoo.yaml \
  --namespace odoo \
  --version 18.2.5
```

The deployment that comes up is: 

```
Welcome to the Bitnami odoo container
Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-odoo
Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-odoo/issues

nami    INFO  Initializing odoo
odoo    INFO  Configuring Odoo database...
postgre INFO  Trying to connect to PostgreSQL server
postgre INFO  Found PostgreSQL server listening at postgresql-ha-pgpool.odoo.svc.cluster.local:5432
postgre INFO  PostgreSQL server listening and working at postgresql-ha-pgpool.odoo.svc.cluster.local:5432
odoo    INFO  Preparing Odoo environment...
Error executing 'postInstallation': User bn_odoo already exists
```

My value files look like this

```
global:
  storageClass: rook-ceph-block-ssd
  postgresql:
    password: XXX
    repmgrUsername: repmgruser
    repmgrPassword: repmgrpassword
    repmgrDatabase: repmgrdatabase
  pgpool:
    adminUsername: postgres
    adminPassword: XXX

postgresql:
  replicaCount: 3
  updateStrategyType: RollingUpdate
  podAnnotations:
    backup.velero.io/backup-volumes: data

pgpool:
  replicaCount: 1
  useLoadBalancing: false
  customUsers:
    usernames: bn_odoo
    passwords: XXX

persistence:
  enabled: true
  accessModes:
    - ReadWriteOnce
  size: 8Gi

service:
  type: LoadBalancer
  port: 5432
  loadBalancerIP:
    192.168.0.YYY
```

for postgresql-ha with XXX being the same password all the time. 

The value file for Odoo is:

```
odooUsername: service
odooEmail: somewhere@someone.com
odooPassword: XXX

persistence:
  enabled: true
  storageClass: rook-cephfs-ssd
  accessMode: ReadWriteMany
  size: 8Gi

service:
  type: ClusterIP

postgresql:
  enabled: false

externalDatabase:
  host:  postgresql-ha-pgpool.odoo.svc.cluster.local
  password: XXX
  port: 5432

ingress:
  enabled: true
  hosts:
    - name: odoo.someone.com
      path: /
      tls: true
      tlsSecret: my-tls
  annotations:
    kubernetes.io/ingress.class: nginx
```

Again XXX is the same password as in the Postgres-ha value file. 

Any ideas?

**Comment by alvneiayu** [comment]
hi @joergfelder 

Postgresql-ha + Odoo is not supported. During the postgresql-ha bootup, you are creating an user (bn_odoo in your case) and adding it inside pgpool. Odoo is trying to create again this user setting up a random password during the boot up but the user is already created. Morever the password is not the same one. 

I created an internal task and added it to the backlog. I am not able to give you an ETA. We will update this thread once we have news.

thanks a lot.

Álvaro

**Comment by marcosbc** [comment]
Hi all, we just released version 19.0.0 of this chart which includes [major changes](https://github.com/bitnami/charts/tree/master/bitnami/odoo#1900) in the initialization logic. In particular, this specific issue should be fixed, as now Odoo allows the database user to be created externally.

Let us know if you continue to find issues.
